src/MergeContext.js

```diff
@@ -287,11 +289,23 @@ class Labels
 
     // adding a previously added or existing label is a no-op
     add(name) {
-        const label = this._find(name);
+        let label = this._find(name);
         if (label)
             label.markForAddition();
-        else
-            this._labels.push(new Label(name, false));
+        else {
+            label = new Label(name, false);
+            this._labels.push(label);
+        }
+        return label;
+    }
+
+    // adds a label and immediately applies it to GitHub
+    async addAndPush(name) {
```

Comment:
Technically, the regular/old add() also pushes, just not immediately -- the push is delayed. The fact that add() delays pushing is mostly irrelevant to its callers -- everything would probably "work" even if add() was immediately pushing. That is why it is OK to leave add() name as it is IMO. Let's clarify the difference of this new method though:

```suggestion
    async addImmediately(name) {
```

Reply:
Done.